plugins/inputs/intel_pmt/README.md

```diff
@@ -0,0 +1,434 @@
+
+- as a filepath – the plugin will use local copies.
+The provided filepath should be an absolute path to `pmt.xml` within
+local copies of XML files from the cloned [Intel-PMT Repository].
+
+## Configuration
+
+```toml @sample.conf
+# Intel Platform Monitoring Technology plugin exposes Intel PMT metrics available through the Intel PMT kernel space.
+# This plugin ONLY supports Linux.
+[[inputs.intel_pmt]]
+  ## URL or filepath to PMT XML within PMT repository.
+  ## Both URL and filepath should be absolute.
+  pmt_source = "https://raw.githubusercontent.com/intel/Intel-PMT/master/xml/pmt.xml"
+  
+  ## Filter metrics by their datatype.
+  ## See the Filtering section in README for more details.
+  ## If empty, all metrics are collected.
+  # datatype_metrics = []
+  
+  ## Filter metrics by their name.
+  ## See the Filtering section in README for more details.
+  ## If empty, all metrics are collected.
+  # sample_metrics = []
+```
+
+## Example Configuration: `pmt_source` as a filepath
+
+This configuration uses a local copy of the XML specification:
+
+```toml
+[[inputs.intel_pmt]]
+  pmt_source = "/home/telegraf/Intel-PMT/xml/pmt.xml"
+```
+
+## Example Configuration: C-State residency and temperature with a datatype metric filter
+
+This configuration allows getting only a subset of metrics
+with the use of a datatype filter:
+
+```toml
+[[inputs.intel_pmt]]
+  pmt_source = "/home/telegraf/Intel-PMT/xml/pmt.xml"
+  datatype_metrics = ["tbandwidth_28b","ttemperature"]
+```
+
+## Example Configuration: C-State residency and temperature with a sample metric filter
+
+This configuration allows getting only a subset of metrics
+with the use of a sample filter:
+
+```toml
+[[inputs.intel_pmt]]
+  pmt_source = "/home/telegraf/Intel-PMT/xml/pmt.xml"
+  sample_metrics = ["C0Residency","C1Residency", "Cx_TEMP"]
+```
+
+## Prerequisites
+
+Minimum Linux kernel version 5.11 with
+
+- the `intel_pmt_telemetry` module loaded (on kernels 5.11-5.14)
+- the `intel_pmt` module loaded (on kernels 5.14+)
+
+Intel PMT is exposed on a limited number of devices, e.g.
+
+- 4th Generation Intel® Xeon® Scalable Processors
+(codenamed Sapphire Rapids / SPR)
+- 6th Generation Intel® Xeon® Scalable Processors
+(codenamed Granite Rapids / GNR)
+
+PMT space is located in `/sys/class/intel_pmt` with `telem` files requiring
+root privileges to read.
+
+### If Telegraf is not running as a root user
+
+By default, the `telem` binary file requires root privileges to be read.
+To avoid running Telegraf as a root, use one of these steps:
+
+- Add read permission for the current user to `telem` files exposed by the
+`intel_pmt` driver
+
+  ```sh
+  sudo chmod o+r /sys/class/intel_pmt/telem*/telem
+  ```
+
+- Add the following capabilities to the Telegraf executable
+
+  ```sh
+  sudo setcap cap_dac_read_search+ep /usr/bin/telegraf
+  ```
+
+## Metrics
+
+All measurements have the following tags:
+
+- `guid` (unique id of an Intel PMT space).
```

Comment:
How many of these unique Ids are there? using an ID as a tag is frowned upon.

Reply:
There are 6 IDs currently. As with time more hardware will support it they will get their own unique IDs. They don't have a name in the XML that we could use instead.
This ID identifies an Intel PMT space, which in turn identifies the device it's reading from.